streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -54,7 +54,19 @@
     private Sensor createSensor;
 
     public ProcessorNode(final String name) {
-        this(name, null, null);
+        this(name, (Processor<KIn, VIn, KOut, VOut>) null, null);
+    }
+
+    public ProcessorNode(final String name,
+                         final Processor<KIn, VIn, KOut, VOut> processor,
+                         final Set<String> stateStores) {
+
+        this.name = name;
+        this.processor = processor;
+        this.children = new ArrayList<>();
+        this.childByName = new HashMap<>();
+        this.stateStores = stateStores;
+        this.time = new SystemTime();
     }
 
     public ProcessorNode(final String name,
```

Comment:
Could we deprecate the old constructor?

Reply:
It will actually get removed later on in the KIP-478 implementation. What will happen is that the DSL and the PAPI will adapt the deprecated Processors to the new ones, and we can completely remove all mentions of the deprecated Processors in the internals.

But for now, if we deprecate this constructor, it'll just obligate us to suppress the deprecation warning in a bunch of other places.